src/main/java/seedu/address/logic/commands/ShowCommand.java

```diff
@@ -0,0 +1,219 @@
+        this(null, id, null);
+    }
+
+    public ShowCommand(Assessment assessment) {
+        this(null, null, assessment);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        return getAssessment().isEmpty() ? showStudent(model) : showAssessment(model);
+    }
+
+    /**
+     * Executes command when a {@code Student} info is requested.
+     */
+    private CommandResult showStudent(Model model) throws CommandException {
+        Predicate<Student> predicate = createStudentPredicate();
+        assert predicate != null;
+
+        model.updateFilteredStudentList(predicate);
+        ObservableList<Student> matchedStudents = model.getFilteredStudentList();
+
+        if (matchedStudents.size() == 0) {
+            throw new CommandException(MESSAGE_NONEXISTENT_STUDENT);
+        }
+
+        if (matchedStudents.size() > 1) {
+            throw new CommandException(MESSAGE_DUPLICATE_STUDENT_NAME);
+        }
+
+        Student matchedStudent = matchedStudents.get(0);
+        Info info = new Info(matchedStudent);
+        return new CommandResult(MESSAGE_SUCCESS, info);
+    }
+
+    /**
+     * Executes command when a {@code Student} info is requested.
+     */
+    private CommandResult showAssessment(Model model) throws CommandException {
+        assert getAssessment().isPresent();
+        Assessment matchedAssessment = model.getAssessment(assessment);
+
+        if (matchedAssessment == null) {
+            throw new CommandException(MESSAGE_NONEXISTENT_ASSESSMENT);
+        }
+
+        matchedAssessment.setStatistics();
+        Info info = new Info(matchedAssessment);
+        AssessmentStatistics statistics = matchedAssessment.getStatistics();
+        return new CommandResult(MESSAGE_SUCCESS, info, statistics.toHistogram());
+    }
+
+    /**
+     * Creates a {@code Predicate} checking if a student has a matched name or ID.
+     */
+    private Predicate<Student> createStudentPredicate() {
+        return name != null ? new NameContainsKeywordsPredicate(Arrays.asList(name.toString().split("\\s+")))
+                : id != null ? new IdContainsKeywordsPredicate(Collections.singletonList(id.toString())) : null;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof ShowCommand)) {
+            return false;
+        }
+
+        // state check
+        ShowCommand toCompare = (ShowCommand) other;
+        boolean isNameEquals = (name != null && name.equals(toCompare.name)) || name == toCompare.name;
+        boolean isIdEquals = (id != null && id.equals(toCompare.id)) || id == toCompare.id;
+        boolean isAssessmentEquals = (assessment != null && assessment.equals(toCompare.assessment))
+                || assessment == toCompare.assessment;
```

Comment:
```suggestion
        boolean isNameEquals = name != null && name.equals(toCompare.name);
        boolean isIdEquals = id != null && id.equals(toCompare.id);
        boolean isAssessmentEquals = assessment != null && assessment.equals(toCompare.assessment);
```

since the respective `.equals` methods already do the reference equality check

Reply:
The reference equality check is for checking the case where both fields of two objects are `null` so I suppose it is necessary?

Reply:
Hmm you're right (sorry!), in this case would 
```java
boolean isNameEquals = name == null ? toCompare.name == null : name.equals(toCompare.name);
```
better convey this intention? Same for the other fields